src/malli/registry.cljc

```diff
@@ -11,12 +11,22 @@
 (defn simple-registry [schemas]
   (reify
     Registry
-    (-schema [_ type] (get schemas type))
+    (-schema [_ type] (schemas type))
     (-schemas [_] schemas)))
 
-(defn registry [?registry]
-  (cond (satisfies? Registry ?registry) ?registry
-        (map? ?registry) (simple-registry ?registry)))
+(defn registry
```

Comment:
simplified this into:

```clj
(defn registry [?registry]
  (cond (nil? ?registry) nil
        #?@(:clj [(instance? malli.registry.Registry ?registry) ?registry])
        (map? ?registry) (simple-registry ?registry)
        (satisfies? Registry ?registry) ?registry))
```

`instance?` -check works with reified classes? woot.

Reply:
They do. I dug into why `satisfies?` is fast for reified classes, and voila!
I also found that `nil?` checks are slightly slower than raw `if`